x-pack/plugins/uptime/public/components/common/header/page_header.tsx

```diff
@@ -36,36 +38,32 @@ const StyledPicker = styled(EuiFlexItem)`
   }
 `;
 
-export const PageHeader = () => {
-  const isCertRoute = useRouteMatch(CERTIFICATES_ROUTE);
-  const isSettingsRoute = useRouteMatch(SETTINGS_ROUTE);
-  const isStepDetailRoute = useRouteMatch(STEP_DETAIL_ROUTE);
-
-  const DatePickerComponent = () =>
-    isCertRoute ? (
-      <CertRefreshBtn />
-    ) : isStepDetailRoute ? null : (
-      <StyledPicker grow={false} style={{ flexBasis: 485 }}>
-        <UptimeDatePicker />
-      </StyledPicker>
-    );
-
-  const isMonRoute = useRouteMatch(MONITOR_ROUTE);
-
+export const PageHeader = ({
```

Comment:
This component was adjusted based off @andrewvc's suggestion that the comment should rely on props for its render state, rather than context about the current route. This refactor allows the component to be more declarative, but is still a bit wonky. The component must have full context of all the possible types of headers, which could become unmanageable.  Another option could be to use children or render props to give even more control of the specifics of what is rendered to each page component, but would require additional refactoring to move that render logic to the page components.